lib/model/alarm_settings.dart

```diff
@@ -40,12 +86,16 @@ class AlarmSettings {
         vibrate = wire.vibrate,
         warningNotificationOnKill = wire.warningNotificationOnKill,
         androidFullScreenIntent = wire.androidFullScreenIntent,
-        allowAlarmOverlap = false;
+        allowAlarmOverlap = wire.allowAlarmOverlap;
 
-  /// Unique identifier assiocated with the alarm. Cannot be 0 or -1;
+  /// Unique identifier associated with the alarm. Cannot be 0 or -1.
   final int id;
 
   /// Date and time when the alarm will be triggered.
+  @JsonKey(
```

Comment:
Why is this needed? IIRC the default behaviour of `JsonSerializable` is identical?

Reply:
For V4 you are defining custom `fromJson` logic in the function above.

If someone upgrades from V4 to V5 we want `toJson` to use the new String format, right?

So I don't think this additional logic is neccesary.

Reply:
You're right, I've done some tests without it and it works fine.